buildingmotif/dataclasses/template.py

```diff
@@ -440,6 +443,52 @@ def find_subgraphs(
         for mapping, sg in matcher.building_mapping_subgraphs_iter():
             yield mapping, sg, matcher.remaining_template(mapping)
 
+    def generate_spreadsheet(self, path: PathLike, inline: bool = False):
```

Comment:
Is it possible to make this support csv as well? And the ability to return the text instead of writing the file. I feel like we need those features in order to connect this with the front end via the API.

Reply:
I updated the API to spit out a buffer if no file path is provided. There's a separate method for writing to the CSV file because I couldn't see a clean way of having the method do both things